packages/entities/entities-plugins/src/components/PluginSelectCard.vue

```diff
@@ -0,0 +1,310 @@
+<template>
+  <KTooltip
+    class="plugin-card plugin-card-cursor-pointer"
+    :class="{
+      'disabled': !plugin.available || plugin.disabledMessage,
+    }"
+    :label="plugin.disabledMessage"
+    position-fixed
+  >
+    <KCard
+      class="plugin-card-content"
+      :data-testid="plugin.name"
+      :disabled="!plugin.available || plugin.disabledMessage"
+      has-hover
+      @click="isCustomPlugin ? undefined : noRouteChange ? emitPluginData() : handleCreateClick()"
```

Comment:
suggestion: instead of the conditional logic inline here, can you map this to a function and then call whatever is appropriate conditionally there instead?